internal/util/fetch/fetch.go

```diff
@@ -21,45 +21,54 @@ import (
 	"os"
 	"os/exec"
 	"path"
-	"path/filepath"
 	"strings"
 
 	"github.com/GoogleContainerTools/kpt/internal/gitutil"
 	"github.com/GoogleContainerTools/kpt/internal/util/git"
+	"github.com/GoogleContainerTools/kpt/internal/util/pkgutil"
 	kptfilev1alpha2 "github.com/GoogleContainerTools/kpt/pkg/api/kptfile/v1alpha2"
 	"github.com/GoogleContainerTools/kpt/pkg/kptfile/kptfileutil"
-	"sigs.k8s.io/kustomize/kyaml/copyutil"
 	"sigs.k8s.io/kustomize/kyaml/errors"
 )
 
-// Command fetches a package from a git repository and copies it to a local
-// directory.
+// Command takes the upstream information in the Kptfile at the provided path,
+// and fetches the package referenced.
 type Command struct {
-	RepoSpec *git.RepoSpec
-
-	Destination string
-
-	Name string
-
-	Clean bool
+	Path string
```

Comment:
Mentioned previously: This should use the Package type.

Per the PR description, it's fine to have this in follow up PR since this is already a large change.

Reply:
Plan on looking at this in a follow-up PR.